crates/server/src/admin/service.rs

```diff
@@ -266,7 +265,7 @@ async fn serve_file(file: &EmbeddedFile) -> Result<impl IntoResponse, StatusCode
     Response::builder()
         .status(StatusCode::OK)
         .header("Content-Type", file.metadata.mimetype())
-        .body(Body::from(file.data.to_vec()))
+        .body(Body::from(file.data.to_owned()))
```

Comment:
quick note:

[`to_owned`](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html#impl-ToOwned-for-%5BT%5D) will clone the slice into a new allocation, no different from [`to_vec`](https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec).

[`into_owned`](https://doc.rust-lang.org/std/borrow/enum.Cow.html#method.into_owned) won't if you already have an owned allocation, which, following the code is likely to be the case

I suspect `into_owned` didn't work because it needs to take ownership of `self`, which is why you fell back to `to_owned`, however, you do in fact have ownership of the `Cow` .._